### ISSUE TITLE
Grant application for the implementation of federation

### ISSUE THREAD

**Issue by pilou-** [go-gitea/gitea#16518]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/grant-application-for-federation-in-gitea/284)
<hr>

Bonjour,

TL;DR: grant application for implementing federation, because it is not making progress (100% transparent)

Despite some [interesting discussions late 2020](https://discourse.gitea.io/t/forgefed-federation-in-gitea/1157), the magnitude of the effort to move forward with the implementation of federation is apparently too much for the developers currently involved in Gitea. It is best demonstrated by the [5K€ grant](https://github.com/go-gitea/gitea/issues/14186#issuecomment-875779897) earmarked for federation: it did not trigger any proposal. Months ago [cjslep](https://github.com/cjslep) suggested applying for a grant (which is an order of magnitude more money) would help and [the discussion resumed a week ago](https://socialhub.activitypub.rocks/t/grant-proposal-for-federation-in-gitea/1921). 

The conclusion  of [this conversation](https://socialhub.activitypub.rocks/t/grant-proposal-for-federation-in-gitea/1921) was to start working on (i) a generic grant application, (ii) use the material of this generic grant application whenever a call is open that would be a good fit. I will do the legwork, reporting progress here the same day. Full disclosure: although I do not intend to be funded by this grant, it is in sync with [the fedeproxy project](https://fedeproxy.eu/) that I am working on; therefore my interest is not purely altruistic. 

The only condition I set for my participation is that **the entire process of the grant application is 100% transparent, as well as the work funded by the grant** (see [the fedeproxy grant to understand exactly what I mean by 100% transparent](https://forum.fedeproxy.eu/t/dapsi-call-deadline-20-01-2021/14)). [@zeripath](https://github.com/zeripath), [cjslep](https://github.com/cjslep) and [techknowlogick](https://github.com/techknowlogick) are aware of this particular requirement and did not raise any concern so far. But if anyone feel like it is too much, please speak up now.

Cheers

**Comment by pilou-** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/grant-application-for-federation-in-gitea/284/2)
<hr>

 Bonjour,

I started working on the generic grant application today and [the result is here](https://demo.hedgedoc.org/K8IDhSxJShOCOWEKILGZeA?view). You're welcome to participate if you'd like. If you find that mildly exciting I won't be offended: it is quite boring.

As for an actual open call, maybe it would make sense to aim for [NGI atlantic call](https://ngiatlantic.eu/ngiatlanticeu-4th-open-call), deadline September 15th, 2021 with a funding range of €25,000 - €75,000. Potential categories that would be a fit are:

* **Topic a: EU – US Experimental Platforms interconnection:** this topic invites established designers and facility providers of experimental infrastructures, testbeds, and platforms enabling experimentation in the NGI areas on both sides of the Atlantic to interconnect with each other, to offer their facilities on a continuous basis to its community of experimenters and application developers in other NGI topics.
* **Topic b.3: Service and data portability**
* **Topic b.5: Internet data sharing and interoperable services** (but that would be streching it too much, IMHO)

That is assuming there exists a counterpart in the US funded by the NSF, which may be a blocker. 

Last but not least, there is a [licensing question to clarify](https://github.com/go-gitea/gitea/issues/14186#issuecomment-884929279) before moving forward.

I'll resume work here next week.

**Comment by pilou-** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/advancing-federation-in-gitea/240/20)

---

@zeripath in [the 1.16 plan](https://github.com/go-gitea/gitea/issues/16429#issuecomment-880042684) you write "(It's worth noting that although this may appear orthogonal to things like federation - this is key to making federation possible.)". You mentioned this a few times and I concur: authentication / authorization is a building block of federated activities. Maybe I'm stating the obvious but here it is: if issues were created around that topic that clearly articulate how they are relevant to federation, it would be possible to fund their implementation with the money earmarked for federation. My 2cts ;-)

**Comment by zeripath** [comment]
I'll try to open appropriate issues as I hit them.

**Comment by pilou-** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/grant-application-for-federation-in-gitea/284/5)

---

Bonjour,

TL;DR: [first rough draft of the generic grant application](https://forum.fedeproxy.eu/uploads/short-url/5kkjPaFehV81rKdomM0TzoI1HbW.md), expecting bio from [zeripath](https://github.com/zeripath) & [cjslep](https://github.com/cjslep)

The essentials of the grant application are [ready](https://demo.hedgedoc.org/K8IDhSxJShOCOWEKILGZeA). It went a lot faster than I expected because it was possible to re-use [the work done by @cj](https://discourse.gitea.io/t/forgefed-federation-in-gitea/1157). It outlines a sensible development plan and is a good starting point. I've added items that [were discussed a few weeks ago](https://socialhub.activitypub.rocks/t/grant-proposal-for-federation-in-gitea/1921) such as reviewing Gitea PRs unrelated to federation. Or spending time to care for diversity. This is what makes a healthy Free Software community and any grant application should explicitly add those deliverables to be sustainable.

As discussed before, the ideal applicants for this grant application would be [cjslep](https://github.com/cjslep) on the go-fed side of things and [zeripath](https://github.com/zeripath), or another core Gitea developer. Although it may not be possible because they are otherwise busy at the time, it would make for a great example if the draft grant application included a description of their background and motivation. Using my background ([3.1 in the fedeproxy grant application](https://forum.fedeproxy.eu/uploads/short-url/67QsDhlh47dpvuJxZf4hAsffZF9.pdf)) and  my motivation ([3.2 in the fedeproxy grant application](https://forum.fedeproxy.eu/uploads/short-url/67QsDhlh47dpvuJxZf4hAsffZF9.pdf)) for inspiration.

When actually applying for a grant, another developer is likely to replace one of them, but only this bio would have to be substituted. And the time spent on the **onboarding** part of the work plan will have to be adjusted accordingly. But the application would otherwise be the same.

[zeripath](https://github.com/zeripath) & [cjslep](https://github.com/cjslep) do you feel like you could write down something?

Now that the generic grant application is drafted, my next action item will be to rewrite it entirely so that it can be something convincing to a skilled Free Software Go developer. If it can convince them the project is doable and worth pursuing, that will be a first major milestone. I don't mean the polite: "this is awesome!" answer from a friend :stuck_out_tongue: I mean the inspired "interesting" answer from a stranger. 

If someone finds the time to read the draft and comment on it, that would be :sparkles: fantastic.

Cheers

**Comment by pilou-** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/advancing-federation-in-gitea/240/23)

---

A grant application has a better chance to be considered if it has the backing of known people/organizations and/or many individuals. Gitea is popular and and effort to get such support is likely to succeed.

1. Is someone willing to advertise this grant proposal and get endorsements and a significant number of individual supporters?
1. What are the communication channels this person should use?
1. What kind of message should they send?

**Comment by pilou-** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/grant-application-for-federation-in-gitea/284/16)

---

Bonjour,

TL;DR: [first version](https://demo.hedgedoc.org/K8IDhSxJShOCOWEKILGZeA?both#) of the generic grant application ready for comments, expecting bio from [zeripath](https://github.com/zeripath) & [cjslep](https://github.com/cjslep).

This [first version](https://demo.hedgedoc.org/K8IDhSxJShOCOWEKILGZeA?both#) should contain all the elements needed to fill a grant application with requirements similar to NLnet and DAPSI (links are included in the document itself if you're curious).

In addition to the [expected bios](https://github.com/go-gitea/gitea/issues/16518#issuecomment-892961905) it would be useful to get a list of organizations or individuals that would [support such a grant application](https://github.com/go-gitea/gitea/issues/16518#issuecomment-892964312).

There are a few question marks in the grant application for which there does not seem to be a publicly available answer but someone from the Gitea community surely does:

* What is the [contribution of the sponsors listed on the gitea.io homepage](https://gitea.io/)?
* Is there an estimate of the number of Gitea installations and/or developers using Gitea?

Next week I'll rework the grant proposal based on comments and double check with the grant guidelines that nothing significant is missing.

Cheers

**Comment by zeripath** [comment]
Hi Loïc! 

Sorry I still haven't provided the bio. Was working on something else this evening - which is hopefully done now.

Will get the bio to you tomorrow.

**Comment by pilou-** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/grant-application-for-federation-in-gitea/284/20)

---

Bonjour,

TL;DR: One last call to finalize the grant application, select your [availability as soon as possible](https://framadate.org/jO19mi38nMKWNYbt)

The [generic grant application](https://demo.hedgedoc.org/K8IDhSxJShOCOWEKILGZeA?both#) is almost done. There are a **few blockers** that can be quickly resolved during a one hour videoconference. Please select your [availability as soon as possible](https://framadate.org/jO19mi38nMKWNYbt), between August 21st and August 28th. Unless you're on vacation :-P

Once this step is complete, applying for a grant will be a matter of cherry-picking. I would not go as far as to say it is easy, but it certainly is a lot easier than starting from scratch :sweat_smile: 

Cheers

**Comment by aschrijver** [comment]
Posted on behalf of [Loic](https://forum.fedeproxy.eu/t/grant-application-for-federation-in-gitea/284/24)

---

> One last call to finalize the grant application, select your [availability as soon as possible](https://framadate.org/jO19mi38nMKWNYbt)

The call is open to everyone and will be at:

**https://meet.jit.si/LargeGuidancesStabilizeCleverly Wednesday August 25th, 6pm UTC+2, 2021**

The call will be recorded and published the same day. Please speak up now if this prevents you from participating. The proposed agenda is:

* Live proofreading of the [executive summary](https://demo.hedgedoc.org/K8IDhSxJShOCOWEKILGZeA?view#Executive-summary): how convincing is it?
* Baby step: [create a private key ](https://forum.fedeproxy.eu/t/advancing-federation-in-gitea/240/26)
* Looking for individuals & organizations willing to support (i) this grant application and/or (ii) advancing federation in Gitea (what is the contributions of the current [sponsors](https://gitea.io/)?).
* Review of the [facts and articles that demonstrate the popularity of Gitea](https://forum.fedeproxy.eu/t/how-popular-is-gitea/332)
* Where to advertise the effort towards federation?
* First grant application could be sent before [October 1st, 2021 for the NGI Zero Discovery](https://nlnet.nl/discovery/) call
